libs/ml/tests/unit/ops/embeddings.cpp

```diff
@@ -203,8 +203,7 @@ TYPED_TEST(EmbeddingsTest, saveparams_test)
   new_op.Forward({std::make_shared<TensorType const>(input)}, new_prediction);
 
   // test correct values
-  EXPECT_TRUE(
-      new_prediction.AllClose(prediction, static_cast<DataType>(0), static_cast<DataType>(0)));
+  EXPECT_TRUE(new_prediction.AllClose(prediction, DataType{0}, DataType{0}));
```

Comment:
ideally all of these static_cast<DataType>(0) -> DataType{0} should be done in a separate PR.
This PR already has to be large and hence tricky to review, but it's made even larger with all these tidying up changes.